pages/docs/couple-your-code/couple-your-code-porting-v2-3.md

```diff
@@ -16,6 +16,85 @@ Please add breaking changes here when merged to the `develop` branch.
 
 ## preCICE API
 
+```diff cpp
+  turnOnSolver(); //e.g. setup and partition mesh
+
+  precice::SolverInterface precice("FluidSolver","precice-config.xml",rank,size); // constructor
+
+- const std::string& coric = precice::constants::actionReadIterationCheckpoint();
+- const std::string& cowic = precice::constants::actionWriteIterationCheckpoint();
+- const std::string& cowid = precice::constants::actionWriteInitialData();
+
+  int dim = precice.getDimension();
+- int meshID = precice.getMeshID("FluidMesh");
+  int vertexSize; // number of vertices at wet surface
+  // determine vertexSize
+  double* coords = new double[vertexSize*dim]; // coords of vertices at wet surface
+  // determine coordinates
+  int* vertexIDs = new int[vertexSize];
+- precice.setMeshVertices(meshID, vertexSize, coords, vertexIDs);
++ precice.setMeshVertices(meshName, vertexSize, coords, vertexIDs);
+  delete[] coords;
+
+- int displID = precice.getDataID("Displacements", meshID);
+- int forceID = precice.getDataID("Forces", meshID);
+  double* forces = new double[vertexSize*dim];
+  double* displacements = new double[vertexSize*dim];
+
+  double solverDt; // solver timestep size
+  double preciceDt; // maximum precice timestep size
+  double dt; // actual time step size
+
+- preciceDt = precice.initialize();
+
+- if(precice.isActionRequired(cowid)){
+-   precice.writeBlockVectorData(forceID, vertexSize, vertexIDs, forces);
+-   precice.markActionFulfilled(cowid);
+- }
++ if(precice.requiresInitialData()){
++   precice.writeBlockVectorData("FluidMesh", "Forces", vertexSize, vertexIDs, forces);
++ }
+
+- precice.initializeData();
++ precice.initialize();
++ precice_dt = precice.getMaxTimeStepSize();
```

Comment:
I would move this call just before the "min".
The diff is then slightly harder to understand, but probably we will already explain details above, so this code could be the "best version".

Reply:
Done. This also allows us to only do a single `getMaxTimeStepSize` call (I will also propagate this change to https://github.com/precice/precice.github.io/pull/258)